osbs/core.py

```diff
@@ -705,6 +705,54 @@ def get_image_stream_tag(self, tag_id):
         check_response(response)
         return response
 
+    def put_image_stream_tag(self, tag_id, tag):
+        url = self._build_url("imagestreamtags/%s" % tag_id)
+        response = self._put(url, data=tag,
+                             headers={"Content-Type": "application/json"})
+        check_response(response)
+        return response
+
+    def ensure_image_stream_tag(self, stream, tag_id, tag_template, scheduled=False):
```

Comment:
I think the API needs to make sense all together, so let's just try to aim for consistency. Something doesn't quite feel right about these two. The calling code shouldn't have to know very much about the details of how the objects are structured, so expecting it to format JSON seems too much.